frontend/src/Settings/index.jsx

```diff
@@ -1,14 +1,130 @@
+    paddingTop: 6,
+    flexDirection: 'row-reverse',
+    alignItems: 'flex-start',
+  },
+  container: {
+    flex: 1,
+    paddingTop: 40,
+    flexDirection: 'column',
+    alignItems: 'center',
+  },
+  background: {
+    width: '100%',
+    height: '100%',
+  },
+  footer: {
+    flex: 1,
+    justifyContent: 'flex-end',
+    marginBottom: 60,
+  },
+  logoutButton: {
+    width: 280,
+  },
+});
 
 function Settings() {
+  const [emailInput, setEmailInput] = useState('');
+  const [usernameInput, setUsernameInput] = useState('');
+  const [kingdomNameInput, setKingdomNameInput] = useState('');
+  const [passwordInput, setPasswordInput] = useState('');
+
+  function handleSubmit() {
+    const settings = {};
+    let message = '';
+    if (emailInput) {
+      settings.email = emailInput;
+      message += 'Your email has changed.\n';
+    }
+    if (usernameInput) {
+      settings.username = usernameInput;
+      message += 'Your username has changed.\n';
+    }
+    if (kingdomNameInput) {
+      settings.kingdom = kingdomNameInput;
+      message += 'Your kingdom\'s name has changed.\n';
+    }
+    if (passwordInput) {
+      settings.password = passwordInput;
+      message += 'Your password has changed.\n';
+    }
+    Alert.alert('Settings', message);
+  }
+
+  const submitButtonIsDisabled = emailInput === '' && usernameInput === '' && kingdomNameInput === '' && passwordInput === '';
+
   return (
-    <View style={[styles.screenStyle,
-      { flex: 1, justifyContent: 'center', alignItems: 'center' }]}
+    <ImageBackground
+      style={styles.background}
+      resizeMode="cover"
+      source={background}
     >
-      <Text>Mocked Settings</Text>
-    </View>
+      <View style={styles.container}>
+        <Text style={styles.titleText}>Settings</Text>
+
+        <Text style={styles.labelText}>email</Text>
+        <InputField
+          placeholder={useSelector((state) => state.settings.email)}
+          value={emailInput}
+          onChangeText={(value) => setEmailInput(value)}
+        />
+
+        <Text style={styles.labelText}>username</Text>
+        <InputField
+          placeholder={useSelector((state) => state.settings.username)}
+          value={usernameInput}
+          onChangeText={(value) => setUsernameInput(value)}
+        />
+
+        <Text style={styles.labelText}>Enter your new Kingdom Name here</Text>
+        <InputField
+          placeholder={useSelector((state) => state.settings.kingdomName)}
+          value={kingdomNameInput}
+          onChangeText={(value) => setKingdomNameInput(value)}
+        />
+
+        <Text style={styles.labelText}>password</Text>
+        <InputField
+          placeholder="password123"
+          value={passwordInput}
+          onChangeText={(value) => setPasswordInput(value)}
+        />
+
+        <View style={styles.buttonRow}>
+          <SubmitButton onPress={handleSubmit} disabled={submitButtonIsDisabled} text="Save" />
+          <SubmitButton text="Cancel" />
```

Comment:
this, and the logout button seems disabled because of the grey text color, please change the text color to white, and use grey only when the button is disabled. I think this is very common in buttons/apps, this is why I'm asking

Reply:
can you please add the functionality too? clicking on the `Cancel` button should take the user back to the `My Kingdom` tab

(I ask you to do this too because the cancel button also works on the `Forgotten password` page, and the ok/submit button on the `Login` and the `Registration` page, I think this functionality is part of this task.)

thanks!

Reply:
when the user gets the 'Your XXX has changed' alert, please redirect them to the `My Kingdom` page 

OR 

you can stay on the `Settings` tab too, but in this case please reset the form!